packages/nc-gui/composables/useExtensions.ts

```diff
@@ -35,14 +35,21 @@ export interface ExtensionManifest {
   }
 }
 
+export interface IKvStore<T> {
+  get(key: string): T | null
+  set(key: string, value: T): Promise<void>
+  delete(key: string): Promise<void>
+  serialize(): Record<string, T>
+}
```

Comment:
Something like following might allow us to store different keys in different types which would be more convenient imo
```
export interface IKvStore<T extends Record<string, any>> {
  get<K extends keyof T>(key: K): T[K] | null;
  set<K extends keyof T>(key: K, value: T[K]): Promise<void>;
  delete<K extends keyof T>(key: K): Promise<void>;
  serialize(): Record<string, T[keyof T]>;
}
```